osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -143,12 +143,28 @@ private void beatmapSetsChanged(object? beatmaps, NotifyCollectionChangedEventAr
                     break;
 
                 case NotifyCollectionChangedAction.Remove:
-                    foreach (var set in oldItems!)
+
+                    int? selectionIndex = null;
```

Comment:
@bdach if possible, could i get a quick smell test on this new logic? i want to get this in ASAP and since i rewrote this portion, would appreciate a second set of eyes to make sure i haven't done anything silly.

Reply:
The logic looks sound logically but in actual testing... it doesn't do what I think it should be doing? My answer to the TODO down there "should this exist in song select instead of here?" is yes because this implementation appears to be completely ignoring the sort mode? It's selecting the closest index item *as specified by the order in `Items` which is not *the actual active sorting order*.